src/stream/src/executor/source/state_table_handler.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::Row;
+use risingwave_common::bail;
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::types::ScalarImpl;
+use risingwave_connector::source::{SplitId, SplitMetaData};
+use risingwave_pb::catalog::Table;
+use risingwave_storage::table::streaming_table::state_table::StateTable;
+use risingwave_storage::StateStore;
+
+use crate::executor::error::StreamExecutorError;
+use crate::executor::StreamExecutorResult;
+
+pub struct SourceStateTableHandler<S: StateStore> {
+    state_store: StateTable<S>,
+}
+
+impl<S: StateStore> SourceStateTableHandler<S> {
+    pub fn from_table_catalog(
+        table_catalog: &Table,
+        store: S,
+        vnodes: Option<Arc<Bitmap>>,
+    ) -> Self {
+        Self {
+            state_store: StateTable::from_table_catalog(table_catalog, store, vnodes),
+        }
+    }
+
+    fn string_to_scalar(rhs: impl ToString) -> ScalarImpl {
```

Comment:
Note that `ToString` is equivalent to `Display`.

```suggestion
    fn string_to_scalar(rhs: impl Into<String>) -> ScalarImpl {
```